### ISSUE TITLE
Automatically choose optimal block size

### ISSUE THREAD

**Issue by YingboMa** [YingboMa/JuliaBLAS.jl#6]
http://www.cs.utexas.edu/users/flame/pubs/flawn74.pdf

**Comment by sjkelly** [comment]
I am hacking around with this. There is a rough sketch on my branch here: https://github.com/sjkelly/JuliaBLAS.jl/tree/sjk/hwloc

Unfortunately I can't get the tests to pass yet, but the Hwloc.jl library seems pretty useful towards this. I'll keep studying that paper and see what I can do.

**Comment by YingboMa** [comment]
Indeed, it would be great to see the integration with Hwloc.jl. Thanks! Could you please open a WIP PR?

**Comment by sjkelly** [comment]
I opened: https://github.com/JuliaBLAS/JuliaBLAS.jl/pull/9
